### PR TITLE
Implement script for diffing fsharp repos

### PR DESCRIPTION
**Pull request by jack-pappas**
I've implement a shell script wrapping ``git diff`` to diff the fsharp/fsharp and Microsoft/visualfsharp repositories and produce a patch. The patch is useful when looking for changes which need to be synced (manually) between the two repositories, and for changes in projects or build scripts that may result in behavioral differences between code compiled from the different repos.

I also as a precaution against committing patches generated by this script, modified the ``.gitignore`` to filter out any ``.patch`` files in the same folder as the script. (I also ignored ``lkg/`` to filter out packages downloaded when building for coreclr.)

tl;dr - I was looking for differences in build scripts between the two repos that could cause the issue noted in #2595, then figured it was something worth automating.

### CODE REVIEW

**Comment by jack-pappas** [comment]
One thing we can also do with this script is diff between fsharp/fsharp or Microsoft/visualfsharp and fsharp/FSharp.Compiler.Service, to help look for any differences there which need to be ported. I'll have to make a few tweaks to this script to support that, but it should be fairly straightforward.

**Comment by KevinRansom** [comment]
@jack-pappas 

Thanks for this, looks like it will be useful.

Kevin

**Comment by dsyme** [comment]
(minor comment: personally I find it easier just to do a github compare or run ``windiff.exe``)
